### DISCUSSION TITLE
Speed info 1MiB/s = 12MiB/s ??

### DISCUSSION THREAD

**Discussion by francisuk1989** [nicotine-plus/nicotine-plus#1696] [General]
Love the idea of the new speed information however, I have noticed if a use i know has 12Mb in speed, It shows as 1MiBs/s instead of 12MiB/ so ...

If a user has only got 56Kb in speeds make this read 56Kb/s
If a user has only got 1Mb in speeds make this read 1MiB/s or 100Kb/s
If a user has only got 5Mb in speeds make this read 5MiB/s  or 500Kb/s
If a user has only got 15Mb in speeds make this read 15MiB/s
If a user has only got 1G in speeds make this read 1G/s (Aimed for people who use Soulseek on seedboxes)

**Comment by mathiascode** [comment]
I don't understand what the message is here. Mb (megabit) is not the same as MiB (mebibyte), and we show the speeds that are sent by the server.

**Reply by francisuk1989** [reply]
I sent you a message on IRC, For anyone whos looking at this, Is regarding this change https://github.com/nicotine-plus/nicotine-plus/commit/10f79cda2575b389bc10f4911e279ab23d711c14

**Reply by mathiascode** [reply]
I made some improvements in https://github.com/nicotine-plus/nicotine-plus/commit/6ba0800c1666d17408f685fbf74bc22af161c2bc

**Comment by hboetes** [comment]
@mathiascode and I made some further improvements in 3f499213f177683a87d2c84de91ac05a2bdf4a4f 
What it boils down to is using 3 digits of information for every speed/size, for example:

- 1.23 KiB/s
- 12.3 KiB/s
- 123 KiB/s

With a little exception for numbers between 999 and 1024 where a precision of 4 digits is required.

I hope you like it.
